addons/mail/models/mail_channel_partner.py

```diff
@@ -66,6 +66,25 @@ def write(self, vals):
         return super(ChannelPartner, self).write(vals)
 
     def unlink(self):
+        for channel_partner in self:
+            notification = {
+                'type': 'rtc_sessions_update',
+                'payload': {
+                    'id': channel_partner.channel_id.id,
+                    'rtcSessions': [('insert-and-unlink', {'id': channel_partner.sudo().rtc_session_ids.id})],
+                },
+            }
+            if channel_partner.partner_id:
+                self.env['bus.bus'].sendmany([(
+                    (self._cr.dbname, 'res.partner', channel_partner.partner_id.id),
+                    notification
+                )])
+
+            if channel_partner.guest_id:
+                self.env['bus.bus'].sendmany([(
+                    (self._cr.dbname, 'mail.guest', channel_partner.partner_id.id),
+                    notification
+                )])
```

Comment:
No `sendmany` inside a loop, this code should have a similar format as `_disconnect` in rtc session.

Actually I think it might be correct to directly call `self.rtc_session_ids._disconnect()` and nothing else (including removing the `unlink()` that is done here below).